src/core/shape_inference/include/stft_shape_inference.hpp

```diff
@@ -46,15 +46,17 @@ std::vector<TRShape> shape_infer(const STFT* op,
     if (signal_shape.rank().is_dynamic()) {
         return {signal_shape};
     } else if (!frame_size || !frame_step) {
-        return {TRShape{signal_shape[0], -1, -1, 2}};
+        return {TRShape{signal_shape[0], TDim(ov::util::dim::inf_bound), TDim(ov::util::dim::inf_bound), 2}};
     }
 
     const auto& frame_size_val = (*frame_size)[0];
     const auto& frame_step_val = (*frame_step)[0];
 
     NODE_SHAPE_INFER_CHECK(op,
                            input_shapes,
-                           0 < frame_size_val && frame_size_val < signal_shape[1].get_interval().get_max_val(),
+                           0 < frame_size_val && (signal_shape[1].is_static()
```

Comment:
Just to check:
If `single_shapep[1]` is `1` and `frame_size_val` is one also the check will fail but in error message there will be 
`... size is 1 but must be in range [1, 1]`

Also check check the error message as `signal_shape[1]` can also be interval value and then in message can be:
 `in range [1, (2,3)]`

Reply:
Shape infer has been updated in different PR, the check is `<=` now.
Currently this PR doesn't change shape_infer files.